modules/hal_nordic/nrfs/backends/nrfs_backend_ipc_service.c

```diff
@@ -128,14 +132,45 @@ static void ipc_sysctrl_ept_recv(const void *data, size_t size, void *priv)
 	}
 }
 
+/**
+ * @brief This function will try to send data directly using ipc service
+ *        In case of errors it will retry if configured.
+ *
+ * @param message Pointer to the buffer to send.
+ * @param size Number of bytes to send.
+ * @retval NRFS_SUCCESS Message sent successfully.
+ * @retval NRFS_ERR_IPC Backend returned error during message sending.
+ */
+static nrfs_err_t nrfs_backend_try_send_directly_over_ipc_service(void *message, size_t size)
+{
+	size_t retry_count = CONFIG_NRFS_SEND_RETRY_MAX_COUNT + 1;
+	int ret = 0;
+
+	do {
+		ret = ipc_service_send(&ipc_cpusys_channel_config.ipc_ept, message, size);
+		if (ret < (int)size) {
+			k_usleep(CONFIG_NRFS_SEND_RETRY_DELAY);
```

Comment:
If there are no retries left, there's no point in sleeping here, so `retry_count` should be checked first.